### PR TITLE
Added changes to integrate with the new Jenkins ci

### PR DESCRIPTION
**Pull request by zelahi**
**Description**

This PR contains the code to trigger the job on the new Jenkins set up at: ci.docker.com/public.  It is removing the reference to `withGithubStatus` in moby and instead it should pick up from our internal jenkins integration setup

Finally the windows pet machines have been set to false, so that we do not test windows builds as there is no communication between the new Jenkins and these windows machines

**How it was done**

I removed references to withGithubStatus and set the job to not run on the windows nodes

**Verification**

Testing this was a little funky.  The way I went about testing these changes was: 
1. Create a Jenkins job here: https://ci.docker.com/public/job/moby-test/job/ze-moby/
2.  Next I created two PRs
    - This PR tests the jobs and makes sure that they run: https://github.com/zelahi/moby/pull/1/files.  The corresponding Jenkins job is here: https://ci.docker.com/public/job/moby-test/job/ze-moby/view/change-requests/job/PR-1/
    - This PR tests to make sure the Jenkins notifications are getting populated: https://github.com/zelahi/moby/pull/2
- The corresponding Jenkins job is here: https://ci.docker.com/public/job/moby-test/job/ze-moby/view/change-requests/job/PR-2/

For the first test, it looks like the `janky` job times out after 3 hours.  We may want to shorten the timeout period for janky but overall the communication with the new Jenkins machines looks to work as expected

**Changelog Description**

Integrate moby/moby ci with ci.docker.com/public

**Cute animal pic**

image

@psftw

### CODE REVIEW

File: Jenkinsfile
Comment (andrewhsu) [inline Jenkinsfile]: `amd64 && ubuntu-1804 && overlay2`

**Review by andrewhsu — APPROVED** [review]
LGTM

This PR is important to get in so we can hookup a jenkins instance to run the `Jenkinsfile`. Removing the `withGuthubStatus` will leave just 1 github status for PR checks hooked up to the `Jenkinsfile` as we migrate to this file. The older PR checks for `janky`, `experimental`, et al. will still be left on during this migration.

**Comment by andrewhsu** [comment]
Would be good to squash before merge.

**Comment by thaJeztah** [comment]
@moby/moby-maintainers PTAL - after this is wired up, there will be some changes in how CI works and is triggered (FYI)

File: Jenkinsfile
Comment (thaJeztah) [inline Jenkinsfile]: Does this mean both RS1 and RS5 are disabled by default?

Comment (zelahi) [reply Jenkinsfile]: For now yes. The new Jenkins infra isn't set up with the windows nodes just yet.  We have to spin up windows pet instances and attach them to the ci.docker.com/public

Comment (andrewhsu) [reply Jenkinsfile]: Yes, for now. After this PR is merged we will get more PRs to enable windows tests so the windows tests are on by default before we disable the old, hand-crafted PR checks we have now for moby/moby PRs.

Comment (andrewhsu) [reply Jenkinsfile]: (currently we gotta get the RS1 and RS5 boxes fixed before they can be activated)

**Review by seemethere — APPROVED** [review]

**Review by tiborvass — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

File: Jenkinsfile
Comment (ddebroy) [reply Jenkinsfile]: cc @StefanScherer

**Pull request merged**
